jpterm.py

```diff
@@ -187,8 +188,21 @@ def _load_input_json(filename):
     elif not os.isatty(sys.stdin.fileno()):
         # If stdin is a pipe, we need read the JSON from
         # stdin and then reset stdin this back to the controlling tty.
-        input_json = json.loads(sys.stdin.read())
-        sys.stdin = open(os.ctermid(), 'r')
+        # Replace fd(0) with tty instead of modifying sys.stdin.
+        # See https://github.com/python/cpython/issues/36029#issuecomment-1093968541 # noqa
+        buf = io.BytesIO()
```

Comment:
That would be great to split this code into a separate function or better a separate object ?

Reply:
@springcomp 
Thanks for your review. I split the code into a separate function.